src/main/java/com/minecolonies/coremod/colony/pvp/AttackingPlayer.java

```diff
@@ -61,12 +60,12 @@ public static boolean isValidAttack(final AbstractEntityCitizen citizen, final C
             return false;
         }
 
-        if (colony.getPermissions().getRank(guardColony.getPermissions().getOwner()) == Rank.HOSTILE)
+        if (colony.getPermissions().getRank(guardColony.getPermissions().getOwner()) == guardColony.getPermissions().getRanks().get(guardColony.getPermissions().HOSTILE_RANK_ID))
```

Comment:
Instead of checking hostile rank, we should just have a hostile action flag now too probably